src/test/java/de/tum/in/www1/artemis/exam/ExamIntegrationTest.java

```diff
@@ -57,9 +58,10 @@
 import de.tum.in.www1.artemis.web.rest.dto.*;
 import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
 
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```

Comment:
Note: Required by `@BeforeAll`. Implications: The class is not re-created for each test. Please raise your hand if this is not acceptable.

Reply:
I like this 👍 I tried to add this change for all tests in general (since creating users seems quite slow, especially on postgres) but this led to problems for some test cases. I think this would be quite a useful test improvement in the future.

Reply:
Thanks :) Yea I was surprised to see such results. I've also tried to also create courses just once, however, some tests modify them, and influence other tests. Also, from what I saw, the impact is quite small (~2s reduction). Just went for the low-hanging fruit.

What we have to check, however, is the behaviour w/ parallel tests. I'll try to check with @laadvo & @DominikRemo regarding this, as I actually have no JUnit experience at all 😅

Reply:
I don't see an issue with parallel test execution here, since we also execute different test classes in parallel, not different Test cases within the same class.

Reply:
I was also thinking about adding this for all tests in the JUnit properties file, but it does not work for all tests and I did not have enough time for more investigation.
Maybe we should open an Issue for this. There is performance improvements and maybe we could enable this for all classes and disable (set lifecycle for methods) in special cases.
I'll have a deeper look into it. For now, this works :)

Reply:
Perfect. Yea, it's really a lot of work to untangle the tests, as they're currently treating the resources as non-shared, as they're re-created in `beforeEach`. What has to be done is essentially:
1) identifying tests which do modifications on shared, non-SUT objects (i.e. for ExamIntegration, exams would be "SUT-objects", while courses are non-SUT). In general, they're mostly just read, not modified.
2) create local versions of these modified objects
3) use these local versions in the tests, not the shared ones

However, there can be hidden dependencies between tests, which only show up when tests are run in a non-deterministic order. I.e. I had issues with one test modifying groups of the course, which caused other tests to fail because of access rights issues.

It's really a different way to write tests, and we have to check if it's worth the effort.